keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,594 @@
+Currently tooling maintained by sig-docs does the following:
+1) Clones k/k
+2) Runs tools to generate cmd/kubeadm/app Cobra reference documentation
+3) Prettyfies the documentation with HTML/CSS
+
+The tooling needs to change to start generating the documentation
+from the k/kubeadm repository. SIG docs must be notified of this
+and collaborate with the kubeadm maintainers when the move is about to happen.
+
+There would not be a need for the tooling to continue building old
+reference documentation from k/k instead, as the k8s.io website reference
+documentation is only generated once before a release happens.
+
+The tools for building k/kubeadm reference docs in k/k must be removed.
+The initial proposal is to not support building reference documentation
+from the tools in the k/kubeadm repository, but might need some discussion.
+Currently the consumers of k/k reference documentation are unknown and the
+documentation is no longer tracked in commits. See:
+https://github.com/kubernetes/kubernetes/tree/master/docs
+
+
+#### Dependency updates
+
+With the kubeadm moves out of k/k, the project will be decoupled
+from the /vendor and /staging dependencies that it currently has in k/k.
+This poses the question of how often the kubeadm project will have to
+update its k8s.io/ dependencies in the go.mod file.
+
+Example of a go.mod file for kubeadm:
+https://github.com/neolit123/kubeadm/blob/66b94f324b63005589bab7132193e8031001ba57/kubeadm/go.mod
+
+The current proposal is to manually update these dependencies right
+after each release or during a release cycle if something is critical.
+The k/k issue tracker and Slack will be the source of signal if something
+has to be updated.
+
+##### Scenarios:
+
+###### A bug in k8s.io/[some-library] is found and fixed in /staging
+or /vendor of k/k
+
+The kubeadm maintainers need to watch for such changes or bug reports
+and update the kubeadm go.mod file. Running the change in CI will confirm
+that it is not regressing.
+
+###### The libraries in k/k/staging have a lot of changes since the
+last k8s release
+
+The kubeadm maintainers need to pick commits to update the kubeadm go.mod
+file to. Running the changes in CI will confirm that they are not regressing.
+
+
+###### Kubernetes released with bug fixes in dependency X, that kubeadm
+did not include
+
+Kubeadm should consider backporting a dependency update depending on the
+severity of the issue.
+If the issue is low severity evaluate if the the backport can be skipped.
+If the issue is high servery consider a backport.
+
+### Risks and Mitigations
+
+#### A portion of the kubeadm backend changes and breaks users
+
+As mentioned in "User story 3", the kubeadm maintainers
+will give no guarantees about the kubeadm backend until it is
+announced as stable. README files in the kubeadm repository
+should denote that.
+
+It is out of scope for this proposal to try to solve the problem.
+
+#### The post-submit job for tagging/branching k/kubeadm fails
```

Comment:
How would the maintainers know that the jobs failed?

Reply:
we can make it possible for the job to email on each failure. the k8s test-grid supports that.

Reply:
updating to mention notifications will be sent to `kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com`